### PR TITLE
Utilites: Add CTAS while dropping some columns

### PR DESCRIPTION
**Pull request by iyerr3**
JIRA: MADLIB-1241

This commit adds function to create a new table from existing table
while dropping some of the columns of the source table.

Closes #282

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/524/

File: src/ports/postgres/modules/utilities/test/utilities.ic.sql_in [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ *//* ----------------------------------------------------------------------- */
+
+-- cleanup
+SELECT cleanup_madlib_temp_tables(quote_ident(current_schema()));
```
[the remark this answers is not in the document]
Comment (iyerr3) [reply src/ports/postgres/modules/utilities/test/utilities.ic.sql_in]: Yes

File: src/ports/postgres/modules/utilities/utilities.py_in [hunk 2]
```diff
@@ -853,6 +857,34 @@ def validate_module_input_params(source_table, output_table, independent_varname
...
+                           cols_to_drop, **kwargs):
+    """ Create copy of table while dropping some of the columns
+    Args:
+        @param source_table str. Name of the source table
+        @param out_table str. Name of the output table
+        @param cols_to_drop str. Comma-separated list of columns to drop
+    """
+    input_tbl_valid(source_table, 'Utilities')
+    output_tbl_valid(out_table, 'Utilities')
+
```
[the remark this answers is not in the document]
Comment (iyerr3) [reply src/ports/postgres/modules/utilities/utilities.py_in]: Added a note in the docs.

File: src/ports/postgres/modules/utilities/utilities.py_in [hunk 3]
```diff
@@ -853,6 +857,34 @@ def validate_module_input_params(source_table, output_table, independent_varname
                                            source_table=source_table))
 # ------------------------------------------------------------------------
 
+
+def create_table_drop_cols(source_table, out_table,
```
[the remark this answers is not in the document]
Comment (iyerr3) [reply src/ports/postgres/modules/utilities/utilities.py_in]: Good point. Added the `MinWarning`

File: src/ports/postgres/modules/utilities/test/utilities.sql_in [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+(12, 'overcast', 72, 90, ARRAY[72, 90], ARRAY['c', 'b'], NULL, 'Play'),
+(13, 'overcast', 81, 75, ARRAY[81, 75], ARRAY['a', 'b'], false, 'Play'),
+(15, NULL, 81, 75, ARRAY[81, 75], ARRAY['a', 'b'], false, 'Play'),
+(16, 'overcast', NULL, 75, ARRAY[81, 75], ARRAY['a', 'd'], false, 'Play'),
+(14, 'rain', 71, 80, ARRAY[71, 80], ARRAY['c', 'b'], true, 'Don''t Play');
+
+SELECT dropcols('dt_golf', 'dt_golf2', '"OUTLOOK", "Cont,features", cat_features');
+-- test if the retained columns are present in output table
+SELECT
+    id, id_2, temperature, humidity, windy, class
```
[the remark this answers is not in the document]
Comment (iyerr3) [reply src/ports/postgres/modules/utilities/test/utilities.sql_in]: Added a test for dropped cols.

File: src/ports/postgres/modules/utilities/test/utilities.sql_in [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    "OUTLOOK" text,
+    temperature double precision,
+    humidity double precision,
+    "Cont,features" double precision[],
+    cat_features text[],
+    windy boolean,
+    class text
+) ;
+
+INSERT INTO dt_golf (id,"OUTLOOK",temperature,humidity,"Cont,features",cat_features, windy,class) VALUES
```
[the remark this answers is not in the document]
Comment (iyerr3) [reply src/ports/postgres/modules/utilities/test/utilities.sql_in]: I've added another column, but I expect all chars requiring a double quote to behave similarly.

**Comment by iyerr3** [comment]
Thanks @kaknikhil for the review. I've updated per the comments. Will merge after your approval.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/531/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/533/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/540/

**Comment by fmcquillan99** [comment]
There is a bit of inconsistency related to the last param `cols_to_drop`

```
SELECT madlib.dropcols(
                    'houses',
                    'houses_out',
                    NULL
);
```
results in `houses_out` = 'houses' , i.e., no-op.

But

```
SELECT madlib.dropcols(
                    'houses',
                    'houses_out'
);
```
results in an error:

```
ERROR:  function madlib.dropcols(unknown, unknown) does not exist
LINE 1: SELECT madlib.dropcols(
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
```

I suggest we either make the param `cols_to_drop` mandatory (preferred) or make the 2 cases above consistent.

Also in the user docs:

1) please make the `dropcols` function name bold like the other utility functions on 
 http://madlib.apache.org/docs/latest/group__grp__utilities.html

2) In the left panel, I would suggest we rename
`Developer Database Functions` to `Utilities`
and rename the top level
`Utility Functions` to `Other Functions`

What to you think?

**Comment by iyerr3** [comment]
@fmcquillan99 Suggested changes have been made in latest commit.

**Comment by fmcquillan99** [comment]
looks like user docs lost the params description for dropcols()

**Comment by iyerr3** [comment]
That's just the way it has to be if it needs to be a link (bold) like the
other functions. There are some other functions in that list that have
parameters and you can get their info by clicking into them. Similarly you
can click drop_cols() to get the parameters info. It's either this or have
the parameters in that same page and make it inconsistent with the other
functions.

**Comment by fmcquillan99** [comment]
ah, i see.  I think it is fine as you have put it.

LGTM

**Pull request closed without merging**